File: LeetCode/3550_Smallest_Index_With_Digit_Sum_Equal_to_Index/Solution.py
from typing import List

class Solution:
    def getDigitSum(self, num: int) -> int:
        return sum([int(value) for value in str(num)])


    def smallestIndex(self, nums: List[int]) -> int:
        for index, num in enumerate(nums):
            if self.getDigitSum(num) == index:
                return index
        return -1

File: LeetCode/3550_Smallest_Index_With_Digit_Sum_Equal_to_Index/Solution.c
int digitSum(int num)
{
    int sum = 0;

    while (num > 0) {
        sum += num % 10;
        num /= 10;
    }

    return sum;
}

int smallestIndex(int *nums, int numsSize)
{
    for (int i = 0; i < numsSize; i++) {
        if (digitSum(nums[i]) == i) {
            return i;
        }
    }

    return -1;
}
